src/medCore/gui/viewContainers/medViewContainer.cpp

```diff
@@ -741,13 +757,37 @@ void medViewContainer::openFromSystem()
     if (path.isEmpty())
         return;
 
-    connect(medDataManager::instance(), SIGNAL(dataImported(medDataIndex,QUuid)), this, SLOT(dataReady(medDataIndex,QUuid)));
-    d->expectedUuid = medDataManager::instance()->importPath(path, true, false);
+    open(path);
+}
 
+void medViewContainer::open(const QString & path)
+{
+    QEventLoop loop;
+    QUuid uuid = medDataManager::instance()->importPath(path, false);
+    d->expectedUuids.append(uuid);
+    connect(medDataManager::instance(), SIGNAL(dataImported(medDataIndex,QUuid)),
+            this, SLOT(open_waitForImportedSignal(medDataIndex,QUuid)));
+    while( d->expectedUuids.contains(uuid)) {
+        loop.processEvents(QEventLoop::ExcludeUserInputEvents);
+    }
     //  save last directory opened in settings.
     medSettingsManager::instance()->setValue("path", "medViewContainer", path);
 }
 
+
+void medViewContainer::open_waitForImportedSignal(medDataIndex index, QUuid uuid)
+{
```

Comment:
Or to make it clearer: in your current code, you need mutexes. In the other version, you don't need them so much because another mechanism is used for the loop BUT mutexes are still a good idea.